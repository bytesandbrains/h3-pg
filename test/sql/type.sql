\pset tuples_only on
\set string '\'801dfffffffffff\''
\set asbigint 576988517884755967
\set hexagon ':string::h3index'
\set pentagon '\'844c001ffffffff\'::h3index'

--
-- TEST operators
--
SELECT :hexagon = :hexagon;
SELECT NOT :hexagon = :pentagon;
SELECT NOT :hexagon <> :hexagon;
SELECT :hexagon <> :pentagon;

--
-- TEST bigint casting
--
SELECT :asbigint = :hexagon::bigint;

SELECT :hexagon = :asbigint::h3index;
