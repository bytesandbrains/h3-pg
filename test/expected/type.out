\pset tuples_only on
\set string '\'801dfffffffffff\''
\set asbigint 576988517884755967
\set hexagon ':string::h3index'
\set pentagon '\'844c001ffffffff\'::h3index'
CREATE TABLE h3_test_type (hex h3index PRIMARY KEY);
INSERT INTO h3_test_type (hex) SELECT * from h3_get_res_0_indexes();
--
-- TEST operators
--
SELECT :hexagon = :hexagon;
 t

SELECT NOT :hexagon = :pentagon;
 t

SELECT NOT :hexagon <> :hexagon;
 t

SELECT :hexagon <> :pentagon;
 t

--
-- TEST b-tree operator class
--
SELECT hex = :hexagon FROM (
    SELECT hex FROM h3_test_type WHERE hex = :hexagon
) q;
 t

--
-- TEST hash operator class
--
SELECT COUNT(hex) = 122 FROM (
    SELECT hex FROM h3_test_type WHERE hex IN (SELECT h3_get_res_0_indexes())
) q;
 t

--
-- TEST bigint casting
--
SELECT :asbigint = :hexagon::bigint;
 t

SELECT :hexagon = :asbigint::h3index;
 t

